sebs/aws/aws.py

```diff
@@ -221,18 +373,28 @@ def create_function(self, code_package: Benchmark, func_name: str) -> "LambdaFun
                 storage_client.upload(code_bucket, package, code_prefix)
 
                 self.logging.info("Uploading function {} code to {}".format(func_name, code_bucket))
-                code_config = {"S3Bucket": code_bucket, "S3Key": code_prefix}
-            ret = self.client.create_function(
-                FunctionName=func_name,
-                Runtime="{}{}".format(
-                    language, self._map_language_runtime(language, language_runtime)
-                ),
-                Handler="handler.handler",
-                Role=self.config.resources.lambda_role(self.session),
-                MemorySize=memory,
-                Timeout=timeout,
-                Code=code_config,
-            )
+                code_config = {"S3Bucket": code_bucket, "S3Key": code_prefix} 
+
+            if container_deployment:
+                package_type = "Image"
+                code_config = {"ImageUri": container_uri}
+
+            create_function_params = {
+                "FunctionName": func_name,
+                "Role": self.config.resources.lambda_role(self.session),
+                "MemorySize": memory,
+                "Timeout": timeout,
+                "PackageType": package_type,
+                "Code": code_config, 
+            }
+
+            if not container_deployment:
+                create_function_params["Runtime"] = "{}{}".format(
+                language, self._map_language_runtime(language, language_runtime) )
+                create_function_params["Handler"] = "handler.handler"
+
+            create_function_params = {k: v for k, v in create_function_params.items() if v is not None}
```

Comment:
I think the code could be a little bit more readable:
- Create generic function params
- Do if/else on container_deployment and change parameters, add them, or delete them depending on the branch.
- Create function

Reply:
Yes, I changed it. Now we add parameters based on container_deployment, and create the function